python/neuroglancer/ingest/tasks.py

```diff
@@ -1,66 +1,348 @@
 import json
 import base64
+import re
+import numpy as np
+import itertools
+import h5py
+from tempfile import NamedTemporaryFile
+import blosc
 
-class Task(object):
-    """
-    A task is defined by an arbitrary blob of data that should be understood by the client.
+from base import QUEUE_NAME, PROJECT_NAME, Storage
+from neuroglancer import chunks
+
+class IngestTask(object):
+    """Ingests and does downsampling.
+       We want tasks execution to be independent of each other, so that no sincronization is
+       required.
+       The downsample scales should be such that the lowest resolution chunk should be able
+       to be produce from the data available.
     """
-    def __init__(self):
-        self._body = None
-        self._info_path = None
-    
+    def __init__(self, chunk_path=None, chunk_encoding=None, info_path=None, fromjson=None, _id=None):
+        self.chunk_path = chunk_path
+        self.chunk_encoding = chunk_encoding
+        self.info_path = info_path
+        self.tag = 'ingest'
+        if fromjson:
+            self.payloadBase64 = fromjson
```

Comment:
It actually uses the property setter. This line is equivalent to `self.payloadBase64(fromjson)`